src/java.net.http/share/classes/java/net/http/HttpRequest.java

```diff
@@ -63,25 +63,35 @@
  * copied and modified many times in order to build multiple related requests
  * that differ in some parameters.
  *
- * <p> The following is an example of a GET request that prints the response
- * body as a String:
- *
+ * <p> The following is an example of an asynchronous POST:
  * {@snippet :
  *   HttpClient client = HttpClient.newHttpClient();
  *
  *   HttpRequest request = HttpRequest.newBuilder()
  *         .uri(URI.create("http://foo.com/"))
+ *         .POST(BodyPublishers.ofString("request body text"))
  *         .build();
  *
  *   client.sendAsync(request, BodyHandlers.ofString())
  *         .thenApply(HttpResponse::body)
  *         .thenAccept(System.out::println)
- *         .join(); }
+ *         .join();
+ *  }
  *
  * <p>The class {@link BodyPublishers BodyPublishers} provides implementations
  * of many common publishers. Alternatively, a custom {@code BodyPublisher}
  * implementation can be used.
  *
+ * <p> The builder can be avoided for simple GET requests as the following example
+ * shows:
+ * {@snippet :
+ *   HttpClient client = HttpClient.newHttpClient();
+ *
+ *   HttpRequest request = HttpRequest.GET("https://www.foo.com/");
+ *   String response = client.send(request, BodyHandlers.ofString()).body();
```

Comment:
I am not sure whether line 91 is such a good idea. Wouldn't you typically first check whether the status is 200 before handling the response? I wonder if we could extend HttpResponse to make dealing with this easier?

Reply:
Good point. We could add a method like `bodyWhen(Predicate<ResponseInfo>)` which would return the body if the predicate is satisfied but throws an exception otherwise.

`String response = client.send(request, BodyHandlers.ofString()).bodyWhen(r -> r.statusCode() == 200);`

Reply:
I've pushed an update for discussion where a new method is added to HttpRespose as follows:

`Optional<T> bodyWhen(Predicate<ResponseInfo> predicate)`

Using Optional is more flexible than explicitly throwing an exception in the method